packages/react-menu/src/components/MenuList/renderMenuList.tsx

```diff
@@ -28,6 +28,7 @@ export const renderMenuList = (state: MenuListState) => {
         setFocusByFirstCharacter,
         hasIcons,
         hasCheckmarks,
+        hasMenuListContext: true,
```

Comment:
This looks a bit strange 🙄

Do I correctly get that the goal is to check is a component is wrapped with `MenuListProvider`? In this case I would suggest to compare with a default value instead:

```ts
const defaultValue = {}
const Context = React.createContext()

//

functio useIsWrapped() {
  return defaultValue !== React.useContext(Context)
}
```

But as we are using context selectors I guess it's more complicated. I am correct?

Reply:
yes, I can't find a way to compare default context using context selector

Reply:
```tsx
function useIsWrapped<Value>(context: Context<Value>) {
  const contextValue = React.useContext(
    (context as unknown) as Context<ContextValue<Value>>
  );

  // "-1" is default value for .version, "Provider" will always set it to "0"
  // This will not handle updates, but updates could happen only if we will change
  // React tree, but it means that components will be remounted
  return contextValue.version.current !== -1;
}
```

I made a quick prototype on CodeSandbox: https://codesandbox.io/s/rough-sea-39ssz?file=/src/App.tsx:241-673
I suggest to find a better name for this hook and place it in `context-selector` package as it could useful for other components